client/extensions/woocommerce/components/address-view/index.js

```diff
@@ -1,51 +1,187 @@
+			state: PropTypes.string,
 			country: PropTypes.string.isRequired,
+			postcode: PropTypes.string,
 		} ),
+		allowedCountries: PropTypes.arrayOf( PropTypes.string ),
+		isEditable: PropTypes.bool,
+		nameLabel: PropTypes.string,
+		onChange: PropTypes.func,
+		onFocus: PropTypes.func,
 	};
 
+	static defaultProps = {
+		address: {
+			name: '',
+			street: '',
+			street2: '',
+			city: '',
+			state: '',
+			country: 'US',
+			postcode: '',
+		},
+		allowedCountries: [
+			'US', 'CA'
+		],
+		isEditable: false,
+		nameLabel: '',
+	}
+
 	constructor( props ) {
 		super( props );
 
-		this.edit = this.edit.bind( this );
+		this.state = {
+			address: props.address,
+		};
 	}
 
-	edit() {
-		//TODO: Add edit functionality
-		return false;
+	getStateFieldLabel = () => {
+		const { country, translate } = this.props;
+
+		if ( 'CA' === country ) {
+			return translate( 'Province' );
+		}
+		return translate( 'State' );
 	}
 
-	render() {
-		const __ = i18n.translate;
+	onChange = ( event ) => {
+		debug( event );
+	}
+
+	renderEditable = () => {
+		const { nameLabel, translate } = this.props;
+		const { city, country, name, postcode, street, street2, state } = this.props.address;
+
+		const countryData = find( Countries, { code: country } );
+		const stateSelectDisabled = Boolean( countryData );
+		// const states = []; // : countryData.states;
+
+		debug( 'countryData=', countryData );
 
 		return (
-			<div>
-				<div className="address-view__address">
-					<p className="address-view__address-name">
-						{ this.props.address.name }
-					</p>
-					<p>{ this.props.address.street }</p>
-					<p>{ this.props.address.city }</p>
-					<p>{ this.props.address.country }</p>
+			<div className="address-view__fields-editable">
+				<FormFieldSet>
+					<FormLabel>{ nameLabel || translate( 'Business Name' ) }</FormLabel>
+					<FormTextInput
+						name="name"
+						onChange={ this.onChange }
+						value={ name }
+					/>
+				</FormFieldSet>
+				<FormFieldSet>
+					<FormLabel>{ translate( 'Street address' ) }</FormLabel>
+					<FormTextInput
+						name="street"
+						onChange={ this.onChange }
+						value={ street }
+					/>
+				</FormFieldSet>
+				<FormFieldSet>
+					<FormTextInput
+						name="street2"
+						onChange={ this.onChange }
+						value={ street2 }
+					/>
+				</FormFieldSet>
+				<div className="address-view__editable-city-state-postcode">
+					<FormFieldSet>
+						<FormLabel>{ translate( 'City' ) }</FormLabel>
+						<FormTextInput
+							name="city"
+							onChange={ this.onChange }
+							value={ city }
+						/>
+					</FormFieldSet>
+					<FormFieldSet>
+						<FormLabel>{ this.getStateFieldLabel() }</FormLabel>
+						<FormSelect
+							disabled={ stateSelectDisabled }
+							name="state"
+							onChange={ this.onChange }
+							value={ state }
+						>
+						</FormSelect>
+					</FormFieldSet>
+					<FormFieldSet>
+						<FormLabel>{ translate( 'Postal code' ) }</FormLabel>
```

Comment:
Hi! I've found a possible matching string that has already been translated **65** times:
`translate( 'Postal Code' )` *ES Score: 16*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/14775-pre-egg-store-address-part-two)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).